crates/sui-graphql-rpc/src/data/pg.rs

```diff
@@ -12,9 +14,12 @@ use diesel::{
 };
 use sui_indexer::indexer_reader::IndexerReader;
 
+use tracing::error;
+
 pub(crate) struct PgExecutor {
     pub inner: IndexerReader,
     pub limits: Limits,
+    pub metrics: Option<Metrics>,
```

Comment:
is there a time when we wouldn't want to have metrics passed in? if it was required and not an option, the use/recording below would be easier.